app/routes/extension/SignerAccount/containers/SignerAccount.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { ga } from 'utils/analytics'
+import { SAFES } from 'utils/analytics/events'
+import Layout from '../components/Layout'
+import { createQrImage } from 'utils/qrdisplay'
+
+class SignerAccount extends Component {
+  componentDidMount = () => {
+    const { account } = this.props
+    createQrImage(
+      document.getElementById('qr-signer-address'),
+      account.secondFA.address,
+      4
+    )
+  }
+
+  componentDidUpdate = (prevProps, prevState) => {
+    const { account } = this.props
+    createQrImage(
+      document.getElementById('qr-signer-address'),
+      account.secondFA.address,
+      4
+    )
+  }
+
+  openEtherScan = async () => {
```

Comment:
I would leave this out. As we don't want the user to be aware of the address of this extension. 

We should shortly discuss with @fmrsabino if we try to "obfuscate" the address to prevent accidental usage

Reply:
Ok after a brief discussion with @rmeissner I think we can do the following.

We have a button with text or icon (for each safe) for "Replace Recovery Phrase".

This would generate a QRCode once clicked. The QRCode can be something like:

```json
{
    "address": "<address>",
    "signature": {
        "v": "<integer>",
        "r": "<string>",
        "s": "<string>"
    }
}
```

Where data for the signature is `sha3("GNO" + "<address>")`

On mobile I'd change the instructions to something like:
```
1. Open the browser extension and select your safe
2. Click on "Replace recovery phrase" on the browser extension.
3. Scan the generated QR code with your phone.
```

@tschubotz

Reply:
I think this makes sense, yes.

Reply:
Another idea would be to sign the `safe_address`. In the case that the user has a "change recovery phrase" button for each safe, we could validate with this on the mobile side that the user is really aware for which safe he is changing the recovery phrase

@germartinez @tschubotz @fmrsabino

Reply:
@germartinez As a reminder `address` should be the checksummed address (as it is in the other payloads)